### PR TITLE
getUserData

### PR DESCRIPTION
**Pull request by aaronli03**
## What
- retrieves user display name and user id from Spotify
- stores data in local storage
- removed check in test for redirection when exchangeToken function is called

## Why
- the application now has access to user display name for display purposes
- the application now has access to user id used for future requests
- redirection is handled in the onMount in the callback

## Impact
- Will customer experience be significantly altered? No
- Does the code change cover more than one task? No

## Testing
- tested locally by running `npm run dev`

### CODE REVIEW

File: src/routes/callback/callback.ts [hunk 1]
```diff
@@ -4,3 +4,13 @@ export function retrieveCode() {
...
+
+export async function storeUserData(name: string, id: string) {
+	const storedValueString: string | null = localStorage.getItem('ViBE');
+	if (storedValueString) {
+		const storedValue = JSON.parse(storedValueString);
+		storedValue.display_name = name;
+		storedValue.user_id = id;
+		localStorage.setItem('ViBE', JSON.stringify(storedValue));
+	}
+}
```
Comment (byjackli) [inline src/routes/callback/callback.ts]: This is a critical part of the app, which means more tests for every possible edge case.

Additionally, need to set up try catch statements. You are learning a lot of technical skills, but lacking in prioritizing user experience - like what happens when user is stranded on a broken site?

1. inform customer when shit hits the fan
2. log it so when shit does hit the fan, yk its real.

File: src/api/spotify/service.ts [hunk 2]
```diff
@@ -3,21 +3,26 @@ import type { Song, Artist, LocalStorage } from '../spotify/types';
...
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${accessToken}`
+			}
+		});
+		if (!response.ok) {
+			console.log('getUserData error');
+		}
+		const responseData = await response.json();
+		(data.display_name = responseData.display_name), (data.user_id = responseData.id);
```
Comment (byjackli) [inline src/api/spotify/service.ts]: whats this syntax mean?

File: src/api/spotify/service.ts [hunk 3]
```diff
@@ -3,21 +3,26 @@ import type { Song, Artist, LocalStorage } from '../spotify/types';
...
+		const accessToken = obj.access_token;
+		const response = await fetch('https://api.spotify.com/v1/me', {
+			headers: {
+				'Content-Type': 'application/json',
+				Authorization: `Bearer ${accessToken}`
+			}
+		});
+		if (!response.ok) {
+			console.log('getUserData error');
+		}
```
Comment (byjackli) [inline src/api/spotify/service.ts]: this log statement is a step in the right direction, see comment for more details https://github.com/byjackli/vibe/pull/41/files#r1292859363

File: src/routes/callback/callback.ts (hunk 1 above)
Comment (aaronli03) [reply src/routes/callback/callback.ts]: will update

File: src/api/spotify/service.ts (hunk 2 above)
Comment (aaronli03) [reply src/api/spotify/service.ts]: will update. I think there was a typo then prettier formatted to that.

File: src/routes/callback/callback.ts [hunk 4]
```diff
@@ -4,3 +4,17 @@ export function retrieveCode() {
 	const code = urlParams.get('code');
 	return code;
 }
+
+export async function storeUserData(name: string, id: string) {
+	try {
+		const storedValueString: string | null = localStorage.getItem('ViBE');
+		if (storedValueString) {
```
Comment (byjackli) [inline src/routes/callback/callback.ts:11]: choose one not both. do some research, and discuss before making switch bc idk which one is better

File: tests/jest/api/auth.test.ts [hunk 5]
```diff
@@ -175,7 +174,6 @@ describe('Auth', () => {
 				user_id: ''
 			})
 		);
-		expect(replaceSpy).toHaveBeenCalledWith(`${url}`);
```
Comment (byjackli) [inline tests/jest/api/auth.test.ts:178]: still need to add more tests

File: src/api/spotify/service.ts [hunk 6]
```diff
@@ -3,22 +3,27 @@ import type { Song, Artist, LocalStorage } from '../spotify/types';
...
+				console.log('getUserData error');
+			}
+			const responseData = await response.json();
+			return responseData;
 		}
-	});
-	if (!response.ok) {
-		console.log('getUserData error');
+	} catch (error) {
+		console.log(error);
```
Comment (byjackli) [inline src/api/spotify/service.ts:24]: nit, you can do `console.error` instead

**Review by byjackli — APPROVED** [review]

**Review by byjackli — APPROVED** [review]
